apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/_components/Filters/LogSourceFilter.tsx

```diff
@@ -0,0 +1,111 @@
+'use client'
+
+import { useCallback, useMemo } from 'react'
+
+import { LogSources } from '@latitude-data/core/browser'
+import { Button, Checkbox, Text } from '@latitude-data/web-ui'
+
+import { FilterButton } from './FilterButton'
+
+const LogSourceLabel: { [key in LogSources]: string } = {
+  [LogSources.API]: 'API',
+  [LogSources.Evaluation]: 'Evaluation',
+  [LogSources.Playground]: 'Playground',
+  [LogSources.User]: 'User',
+  [LogSources.SharedPrompt]: 'Shared Prompt',
+}
+
+function LogSourceCheckbox({
+  logSource,
+  selectedLogSources,
+  setSelectedLogSources,
+}: {
+  logSource: LogSources
+  selectedLogSources: LogSources[]
+  setSelectedLogSources: (selectedLogSources: LogSources[]) => void
+}) {
+  const isSelected = useMemo(
+    () => selectedLogSources.includes(logSource),
+    [selectedLogSources, logSource],
+  )
+
+  const onSelect = useCallback(() => {
+    setSelectedLogSources(
+      isSelected
+        ? selectedLogSources.filter((origin) => origin !== logSource)
+        : [...selectedLogSources, logSource],
+    )
+  }, [selectedLogSources, logSource, isSelected])
+
+  return (
+    <Button variant='ghost' fullWidth className='p-0' onClick={onSelect}>
+      <div className='flex flex-row w-full justify-start gap-2'>
+        <Checkbox checked={isSelected} fullWidth={false} />
+        <Text.H5 noWrap ellipsis>
+          {LogSourceLabel[logSource]}
+        </Text.H5>
+      </div>
```

Comment:
A checkbox inside a button? Is a bit weird no? Why not use `<Checkbox label=''>`?